src/mpid/ch4/src/ch4r_recv.h

```diff
@@ -73,10 +73,14 @@ static inline int MPIDIG_handle_unexp_mrecv(MPIR_Request * rreq)
     MPIR_Datatype_get_size_macro(datatype, dt_sz);
 
     if (message_sz > count * dt_sz) {
-        rreq->status.MPI_ERROR = MPI_ERR_TRUNCATE;
+        rreq->status.MPI_ERROR = MPIR_Err_create_code(rreq->status.MPI_ERROR,
+                                                      MPIR_ERR_RECOVERABLE, __FUNCTION__, __LINE__,
+                                                      MPI_ERR_TRUNCATE, "**truncate",
+                                                      "**truncate %d %d %d %d",
+                                                      rreq->status.MPI_SOURCE, rreq->status.MPI_TAG,
+                                                      dt_sz * count, message_sz);
         nbytes = dt_sz * count;
     } else {
-        rreq->status.MPI_ERROR = MPI_SUCCESS;
         nbytes = message_sz;
```

Comment:
Is it OK to do not reset `rreq->status.MPI_ERROR` to `MPI_SUCCESS` ?

Reply:
I think this line is redundant since `MPI_ERROR` is already initialized to `MPI_SUCCESS`, but we can leave it in if you want.

Reply:
If it is already initialized somewhere, we can delete this line.